### PR TITLE
feat: adds uds state

### PR DESCRIPTION
**Pull request by UncleGedd**
## Description

adds UDS state

## Related Issue
https://github.com/defenseunicorns/uds-cli/issues/614

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [ ] Test, docs, adr added or updated as needed
- [ ] [Contributor Guide Steps](https://github.com/defenseunicorns/uds-cli/blob/main/CONTRIBUTING.md) followed

### CODE REVIEW

File: src/pkg/state/state.go [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+				},
+				Data: map[string][]byte{
+					"data": {},
+				},
+			}
+			_, err = c.client.CoreV1().Secrets(stateNs).Create(context.TODO(), stateSecret, metav1.CreateOptions{})
+			if err != nil {
+				return nil, err
+			}
+		}
```
Comment (decleaver) [inline src/pkg/state/state.go:121]: Is it possible to get another error here that is not a "IsNotFound"?

File: src/pkg/bundle/deploy.go [hunk 2]
```diff
@@ -56,23 +58,55 @@ func (b *Bundle) Deploy() error {
...
+	}
+
+	err = sc.InitBundleState(b.bundle)
+	if err != nil {
+		return err
+	}
+
+	// update bundle state with deploying
+	err = sc.UpdateBundleState(b.bundle.Metadata.Name, state.Deploying)
+
```
Comment (decleaver) [inline src/pkg/bundle/deploy.go:78]:
```suggestion
if err != nil {
		return err
	}
```

File: src/pkg/bundle/remove.go [hunk 3]
```diff
@@ -106,8 +152,21 @@ func removePackages(packagesToRemove []types.Package, b *Bundle) error {
 			if err := pkgClient.Remove(context.TODO()); err != nil {
 				return err
 			}
+
+			err = sc.UpdateBundlePkgState(b.bundle.Metadata.Name, pkg.Name, state.Removed)
```
Comment (decleaver) [inline src/pkg/bundle/remove.go:156]:
```suggestion
			err = sc.UpdateBundlePkgState(b.bundle.Metadata.Name, pkg.Name, state.Removed)
			if err != nil {
				return err
			}
```

File: src/pkg/bundle/remove.go [hunk 4]
```diff
@@ -106,8 +152,21 @@ func removePackages(packagesToRemove []types.Package, b *Bundle) error {
 			if err := pkgClient.Remove(context.TODO()); err != nil {
 				return err
```
Comment (decleaver) [inline src/pkg/bundle/remove.go:153]: if we run into an error while removing, is it ok for the state to remain state.Removing?

File: src/pkg/bundle/deploy.go (hunk 2 above)
Comment (UncleGedd) [reply src/pkg/bundle/deploy.go:78]: Good catch, updated!

File: src/pkg/state/state.go (hunk 1 above)
Comment (UncleGedd) [reply src/pkg/state/state.go:127]: yes updated

File: src/pkg/bundle/remove.go (hunk 4 above)
Comment (UncleGedd) [reply src/pkg/bundle/remove.go:153]: We could add another state called `FailedRemove`

Comment (UncleGedd) [reply src/pkg/bundle/remove.go]: 👍 added

File: src/pkg/bundle/remove.go (hunk 3 above)
Comment (UncleGedd) [reply src/pkg/bundle/remove.go:160]: fixed!

**Review by decleaver — APPROVED** [review]
tested out locally, looks good!

**Pull request merged**